src/js/controllers/mapController.ts

```diff
@@ -1142,6 +1145,295 @@ export class MapController {
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (!layer.sublayers) {
+      return;
+    }
+
+    const sublayer24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    sublayer24.visible = false;
+
+    if (layer.id.includes('VIIRS')) {
+      const VIIRS48 = this._map?.findLayerById('VIIRS48');
+      const VIIRS72 = this._map?.findLayerById('VIIRS72');
+      const VIIRS7D = this._map?.findLayerById('VIIRS7D');
+
+      VIIRS48.visible = false;
+      VIIRS72.visible = false;
+      VIIRS7D.visible = false;
+    } else if (layer.id.includes('MODIS')) {
+      const MODIS48 = this._map?.findLayerById('MODIS48');
+      const MODIS72 = this._map?.findLayerById('MODIS72');
+      const MODIS7D = this._map?.findLayerById('MODIS7D');
+
+      MODIS48.visible = false;
+      MODIS72.visible = false;
+      MODIS7D.visible = false;
+    }
+  }
+
+  updateMODISorVIIRSOpacity(layerID: string, opacity: number): void {
+    if (!this._map) {
+      return;
+    }
+
+    const layer = (this._map.allLayers as any).items.filter(
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (!layer.sublayers) {
+      return;
+    }
+
+    const sublayer24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    sublayer24.opacity = opacity;
+
+    if (layerID.includes('VIIRS')) {
+      const VIIRS48 = this._map?.findLayerById('VIIRS48');
+      const VIIRS72 = this._map?.findLayerById('VIIRS72');
+      const VIIRS7D = this._map?.findLayerById('VIIRS7D');
+
+      VIIRS48.opacity = opacity;
+      VIIRS72.opacity = opacity;
+      VIIRS7D.opacity = opacity;
+    } else if (layerID.includes('MODIS')) {
+      const MODIS48 = this._map?.findLayerById('MODIS48');
+      const MODIS72 = this._map?.findLayerById('MODIS72');
+      const MODIS7D = this._map?.findLayerById('MODIS7D');
+
+      MODIS48.opacity = opacity;
+      MODIS72.opacity = opacity;
+      MODIS7D.opacity = opacity;
+    }
+  }
+
+  setDefinedDateRange(layerID: string, sublayerType: string): void {
+    if (!this._map) {
+      return;
+    }
+
+    const layer = (this._map.allLayers as any).items.filter(
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (layerID.includes('MODIS')) {
+      this.setMODISDefinedRange(layer, sublayerType);
+    }
+
+    if (layerID.includes('VIIRS')) {
+      this.setVIIRSDefinedRange(layer, sublayerType);
+    }
+
+    // * Below is an approach we can use if we update the API
+    // * leaving this in case we update the API
+    // layer.sublayers.items.forEach((sublayer: Sublayer) => {
+    //   // * Turns off all sublayers associated
+    //   // * with the selected layer (MODIS OR VIIRS)
+    //   if (!sublayer.title.includes(sublayerType)) {
+    //     sublayer.visible = false;
+    //   } else {
+    //     sublayer.visible = true;
+    //   }
+    // });
```

Comment:
I dont know if we need this, I doubt API will change and if it does by some chance in the future I think we can figure this out.